src/clr-addons/history/history.service.ts

```diff
@@ -0,0 +1,138 @@
+        element =>
+          element.username === username &&
+          element.applicationName === application &&
+          element.tenantid === this.tenantPrefix + tenant
+      );
+    }
+    return history.filter(element => element.username === username && element.applicationName === application);
+  }
+
+  /**
+   * Set history for username, application and tenant id
+   * Keep other untouched
+   * @param entries
+   */
+  private setHistory(entries: ClrHistoryModel[]): void {
+    if (!entries) {
+      // clear all entries
+      this.setCookie(this.cookieName, '');
+    }
+    this.setCookie(this.cookieName, JSON.stringify(entries));
+  }
+
+  resetHistory(): void {
+    this.setHistory(null);
+  }
+
+  removeFromHistory(entry: ClrHistoryModel): void {
+    let history = this.getHistory(entry.username, entry.applicationName, entry.tenantid);
+    /* remove current entry from array */
+    history = history.filter(
+      element =>
+        !(
+          element.pageName === entry.pageName &&
+          element.context.indexOf(entry.context) === 0 &&
+          element.tenantid === this.tenantPrefix + entry.tenantid &&
+          element.username === entry.username
+        )
+    );
+    this.setHistory(history);
+  }
+
+  getHistoryPinned(username: string): boolean {
+    let historySettings: ClrHistorySettingsModel[] = this.getCookieByName(this.cookieNameSettings);
+    if (!historySettings || historySettings.length === 0) {
+      this.setHistoryPinned(username, false);
+      historySettings = [{ username: username, historyPinned: false }];
+    }
+    this.cookieSettings$.next(historySettings);
+    const historySetting = historySettings.find(hSetting => hSetting.username === username);
+    if (historySetting) {
+      return historySetting.historyPinned;
+    }
+    return false;
+  }
+
+  setHistoryPinned(username: string, pin: boolean): void {
+    const historySettings: ClrHistorySettingsModel[] = this.getCookieByName(this.cookieNameSettings);
+    const historySetting = historySettings.find(hSetting => hSetting.username === username);
+    if (historySetting) {
+      historySetting.historyPinned = pin;
+    } else {
+      historySettings.push({ username: username, historyPinned: pin });
+    }
+    this.cookieSettings$.next(historySettings);
+    // Set it
+    this.setCookie(this.cookieNameSettings, JSON.stringify(historySettings));
+  }
+
+  private getCookieByName(name: string): any[] {
+    return (
+      document.cookie
+        .split(';')
+        .map(c => c.trim())
+        .filter(cookie => {
+          return cookie.substring(0, name.length + 1) === `${name}=`;
+        })
+        .map(cookie => {
+          return JSON.parse(cookie.substring(name.length + 1));
+        })[0] || []
+    );
+  }
+
+  private setCookie(name: string, content: string): void {
+    document.cookie =
+      name +
+      '=' +
+      content +
+      ';domain=' +
+      window.location.hostname.split('-').slice(-1) +
```

Comment:
won't work in production, as links look like that over there:
https://outputmanager.cross3.at/crossng-outputmanager/outputmanager